bootstrap/kubeadm/api/v1alpha2/conversion.go

```diff
@@ -39,6 +39,7 @@ func (src *KubeadmConfig) ConvertTo(dstRaw conversion.Hub) error {
 	dst.Status.DataSecretName = restored.Status.DataSecretName
 	dst.Spec.Verbosity = restored.Spec.Verbosity
 	dst.Spec.UseExperimentalRetryJoin = restored.Spec.UseExperimentalRetryJoin
+	dst.Spec.Files = restored.Spec.Files
```

Comment:
I think this is what we want? Otherwise we have to match files based on some property field to match array items?

Reply:
We should only restore the files that have `ContentFrom` populated (non-nil)

Reply:
Is this because the restored version might be stale where conversion was possible?

Reply:
Correct, in case a v1alpha2 controller modified some of the files (even though unlikely because bootstrap should only happen once), we need to make sure to restore those that have the new schema.

Reply:
I'm not sure I follow what this is doing... we're setting `dst.Spec.Files` to  `restored.Spec.Files` and then appending some of the restored files to `dst.Spec.Files`? Isn't that going to result in duplicates?

Reply:
Nice catch.

I think the first line `dst.Spec.Files = restores.Spec.Files` is leftover from my first revision, it should only be the loop. The intent is to look at restored files with contentFrom and append them. We shouldn’t append restored files with content set.

Will PR shortly